### DISCUSSION TITLE
Questions from  markh2016

### DISCUSSION THREAD

**Discussion by keith-packard** [keith-packard/snek#80] [Q&A]
(I'm afraid I hadn't enabled discussions somehow, so let's see if this helps in the future!)

@markh2016 asks in #38 :

Hi

I'm new to this and just started

Ive read through the manual but have a questions

How do I write to serial port as I would with arduino C++

How also do I address pins for lets say writing to a LCD in nibble mode , byte mode using the snek programming language Could you perhaps provide some examples I could follow

Thanking you

Mark Harrington

**Comment by keith-packard** [comment]
The default serial port is generally connected directly to the REPL, so you can use `print` to send data there. If you want to use a different serial port, we'd need to invent a new mechanism for using them, there isn't anything available yet.

To address multiple pins, you'd need to write  a function that addresses them individually; there's no mechanism for setting multiple pins at the same time.

**Reply by markh2016** [reply]
Hi Keith 

Thank you  for your answers 

Will there be any plans for the future to address multiple port pins This would be a very hand feature

**Reply by keith-packard** [reply]
I don't have any personal plans for this feature, but would be happy to review proposed patches. It might be a bit tricky to get right across a range of devices as they don't always make the pin configurations contiguous between the device ports and the pin numbers though. It would be pretty simple for atmega devices where the mapping is simple though.

**Reply by markh2016** [reply]
That would be brilliant for atmel devices 

Thanking you for your answers
